### PR TITLE
Output runtime in statepoint files

### PR DESCRIPTION
**Pull request by smharper**
This short PR adds the output from our various timers to the statepoint file.  This will be handy for checking out the runtime penalties of various input choices, new code methods.

### CODE REVIEW

File: docs/source/usersguide/output/statepoint.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@
 State Point File Format
 =======================
 
-The current revision of the statepoint file format is 15.
+The current revision of the statepoint file format is 16.
```
Comment (paulromano) [inline docs/source/usersguide/output/statepoint.rst]: I'm of the opinion that we only need to increment revisions at most once between releases.

File: docs/source/usersguide/output/statepoint.rst [hunk 2]
```diff
@@ -260,3 +260,69 @@ if (run_mode == 'k-eigenvalue' and source_present > 0)
         ``wgt``, ``xyz``, ``uvw``, ``E``, ``g``, and ``delayed_group``, which
         represent the weight, position, direction, energy, energy group, and
         delayed_group of the source particle, respectively.
+
+**/runtime/total initialization** (*double*)
+
+    Time (in seconds on the master processor) spent reading inputs, allocating
```
Comment (paulromano) [inline docs/source/usersguide/output/statepoint.rst]: "master process" is more accurate than "master processor"

File: docs/source/usersguide/output/statepoint.rst [hunk 3]
```diff
@@ -260,3 +260,69 @@ if (run_mode == 'k-eigenvalue' and source_present > 0)
...
+
+**/runtime/inactive batches** (*double*)
+
+    Time (in seconds on the master processor) spent in the inactive batches
+    (including non-transport activities like communcating sites).
+
+**/runtime/active batches** (*double*)
+
+    Time (in seconds on the master processor) spent in the active batches
+    (including non-transport activities like communcating sites).
```
Comment (paulromano) [inline docs/source/usersguide/output/statepoint.rst]: typo: communicating

File: openmc/statepoint.py [hunk 4]
```diff
@@ -312,6 +316,13 @@ def run_mode(self):
         return self._f['run_mode'].value.decode()
 
     @property
+    def runtime(self):
+        out = dict()
+        for key in self._f['runtime'].keys():
+            out[key] = self._f['runtime/' + key].value
+        return out
```
Comment (paulromano) [inline openmc/statepoint.py]: Could do a one-liner here:
`return {name: dataset.value for name, dataset in self._['runtime'].items()}`

File: src/summary.F90 [hunk 5]
```diff
@@ -717,63 +717,4 @@ subroutine write_tallies(file_id)
 
   end subroutine write_tallies
 
-!===============================================================================
-! WRITE_TIMING
-!===============================================================================
-
-  subroutine write_timing(file_id)
```
Comment (paulromano) [inline src/summary.F90]: About time we got rid of this! Thanks!

**Comment by paulromano** [comment]
Looks good to go pending responses to my comments

**Comment by smharper** [comment]
Comments addressed

**Pull request merged**